Dependencies/Utility/Utility/endian_compat.h

```diff
@@ -0,0 +1,247 @@
+
+#define be32toh(x) betoh32(x)
+#define le32toh(x) letoh32(x)
+
+#define be64toh(x) betoh64(x)
+#define le64toh(x) letoh64(x)
+
+#elif defined(_WIN32) || defined(_WIN64)
+#if !(defined(_MSC_VER) && _MSC_VER < 1300)
+#include <intrin.h>
+#define htobe16(x) _byteswap_ushort(x)
+#define htole16(x) (x)
+#define be16toh(x) _byteswap_ushort(x)
+#define le16toh(x) (x)
+
+#define htobe32(x) _byteswap_ulong(x)
+#define htole32(x) (x)
+#define be32toh(x) _byteswap_ulong(x)
+#define le32toh(x) (x)
+
+#define htobe64(x) _byteswap_uint64(x)
+#define htole64(x) (x)
+#define be64toh(x) _byteswap_uint64(x)
+#define le64toh(x) (x)
+#else
+#define bswap16(x) ( (uint16_t)( ((x & 0xFF00) >>  8) | ((x & 0x00FF) <<  8) ) )
+#define bswap32(x) ( ((x & 0xFF000000) >> 24) | \
+					 ((x & 0x00FF0000) >>  8) | \
+					 ((x & 0x0000FF00) <<  8) | \
+					 ((x & 0x000000FF) << 24) )
+#define bswap64(x) ( (((uint64_t)(x) & 0xFF00000000000000) >> 56) | \
+                     (((uint64_t)(x) & 0x00FF000000000000) >> 40) | \
+                     (((uint64_t)(x) & 0x0000FF0000000000) >> 24) | \
+                     (((uint64_t)(x) & 0x000000FF00000000) >>  8) | \
+                     (((uint64_t)(x) & 0x00000000FF000000) <<  8) | \
+                     (((uint64_t)(x) & 0x0000000000FF0000) << 24) | \
+                     (((uint64_t)(x) & 0x000000000000FF00) << 40) | \
+                     (((uint64_t)(x) & 0x00000000000000FF) << 56) )
+
+#define htobe16(x) bswap16(x)
+#define htole16(x) (x)
+#define be16toh(x) bswap16(x)
+#define le16toh(x) (x)
+
+#define htobe32(x) bswap32(x)
+#define htole32(x) (x)
+#define be32toh(x) bswap32(x)
+#define le32toh(x) (x)
+
+#define htobe64(x) bswap64(x)
+#define htole64(x) (x)
+#define be64toh(x) bswap64(x)
+#define le64toh(x) (x)
+
+#endif // _MSC_VER
+
+#else
+#error platform not supported
+#endif
+
+
+// Endian helper function data types
+#if defined(__linux__) || defined(__CYGWIN__)
+typedef uint16_t SwapType16;
+typedef uint32_t SwapType32;
+typedef uint64_t SwapType64;
+
+#elif defined(__APPLE__)
+typedef UInt16 SwapType16;
+typedef UInt32 SwapType32;
+typedef UInt64 SwapType64;
+
+#elif defined(__OpenBSD__)
+typedef uint16_t SwapType16;
+typedef uint32_t SwapType32;
+typedef uint64_t SwapType64;
+
+#elif defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__)
+typedef uint16_t SwapType16;
+typedef uint32_t SwapType32;
+typedef uint64_t SwapType64;
+
+#elif defined(_WIN32) || defined(_WIN64)
+typedef uint16_t SwapType16;
+typedef uint32_t SwapType32;
+typedef uint64_t SwapType64;
+
+#else
+#error platform not supported
+#endif
+
+
+// Endian helper functions
+static_assert(sizeof(SwapType16) == 2, "expected size does not match");
+static_assert(sizeof(SwapType32) == 4, "expected size does not match");
+static_assert(sizeof(SwapType64) == 8, "expected size does not match");
+
+// VC6 compatible overloaded endian functions
+#if defined(_MSC_VER) && _MSC_VER < 1300
+
+// Big endian to hardware
```

Comment:
Minor: I think h refers to host. See MacOS function name: OSSwapLittleToHostInt16